src/main/java/ru/hogwards/school/controller/FacultyController.java

```diff
@@ -53,4 +54,25 @@ public ResponseEntity deleteFaculty(@PathVariable Long id) {
     public List<Faculty> filterFacultiesByColor(@RequestParam String color) {
         return facultyService.filterByColor(color);
     }
+
+    @GetMapping
+    public ResponseEntity<Faculty> findFacultyByNameOrColor(@RequestParam(required = false) String name,
+                                                            @RequestParam(required = false) String color) {
+        if (name != null && !name.isBlank()) {
+            return ResponseEntity.ok(facultyService.findFacultyByName(name));
+        }
+        if (color != null && !color.isBlank()) {
+            return ResponseEntity.ok(facultyService.findFacultyByColor(color));
+        }
+        return ResponseEntity.badRequest().build();
+    }
+
+    @GetMapping("/students/{id}")
```

Comment:
с точки зрения REST правильнее было бы сначала указать id ресурса, а потом уже то, что от него нужно получить

```suggestion
    @GetMapping("/{id}/students")
```